app/models/invoice.rb

```diff
@@ -32,18 +32,32 @@ class Invoice < ActiveRecord::Base
   validates :ends_at, presence: true
   validates :amount, presence: true
   validates :currency, presence: true
+  validates :customer_name, presence: true
 
   validates :amount, numericality: { greater_than: 0,
     less_than_or_equal_to: Dictionaries.money_max }
   validates :currency, inclusion: { in: Dictionaries.currencies,
     message: "%{value} is not a valid currency" }
+  validate :check_overlap_dates_for_customer, if: 'customer_name.present?'
```

Comment:
@yunixon please check this gem https://github.com/robinbortlik/validates_overlap I think it will be better to use this, instead of custom solution.